src/Commands/Domain/SetPrimaryCommand.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+
+namespace Pantheon\Terminus\Commands\Domain;
+
+use Consolidation\AnnotatedCommand\AnnotationData;
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Commands\WorkflowProcessingTrait;
+use Pantheon\Terminus\Models\Environment;
+use Pantheon\Terminus\Models\Site;
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class SetPrimaryCommand
+ * @package Pantheon\Terminus\Commands\Domain
+ */
+class SetPrimaryCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+    use WorkflowProcessingTrait;
+
+    /**
+     * Sets the primary domain for a site and environment, causing all traffic to redirect to the primary domain.
+     *
+     * @authorize
+     *
+     * @command domain:primary:set
+     *
+     * @param string $site_env Site & environment in the format `site-name.env`
+     * @param string $domain A domain that has been associated to your site. Optional when running interactively.
+     *
+     * @usage domain:primary:set <site_env>
+     */
+    public function set($site_env, $domain)
+    {
+        /**
+         * @var $site Site
+         * @var $env Environment
+         */
+        list(, $env) = $this->getSiteEnv($site_env);
+
+        // The primary domain is set via a workflow so as to use workflow logging to track changes.
+        $this->log()->notice('Setting primary domain to {domain}...', ['domain' => $domain]);
+        $workflow = $env->setPrimaryDomain($domain);
+        $this->processWorkflow($workflow);
```

Comment:
I think it would a good idea to print some sort confirmation message here (e.g. "primary domain was set to abc.com"). We show the progress bar but most often that is quite fast and at the end we're just left with `[notice] Setting primary domain to joytherapy.ro...`. I'm not sure what the status of the operation is at this point.

Reply:
Makes sense to me, this style just seemed to be what other workflow commands were doing :man_shrugging: